test/unit/routes/claim/test-file-upload.js

```diff
@@ -25,16 +49,62 @@ describe('routes/first-time/eligibility/claim/file-upload', function () {
     request = supertest(app)
   })
 
-  describe(`GET ${ROUTE}`, function () {
+  describe(`GET ${BASEROUTE}`, function () {
+    it('should call the CSRFToken generator', function () {
+      request
+        .get(VALIDROUTE)
+        .expect(function () {
+          sinon.assert.calledOnce(generateCSRFTokenStub)
+        })
+    })
+
     it('should respond with a 200 if passed valid document type', function () {
       request
-        .get(`${ROUTE}VISIT_CONFIRMATION`)
+        .get(VALIDROUTE)
         .expect(200)
     })
 
+    it('should call the directory check', function () {
+      request
+        .get(VALIDROUTE)
+        .expect(function () {
+          sinon.assert.calledOnce(directoryCheckStub)
+        })
+    })
+
+    it('should respond with a 500 if passed invalid document type', function () {
+      request
+        .get(`${BASEROUTE}VISIT_CONFIRMATION&claimDocumentId=${CLAIMDOCUMENTID}&eligibilityId=${ELIGIBILITYID}`)
+        .expect(500)
+    })
+  })
+
+  describe(`POST ${BASEROUTE}`, function () {
+    it('should create a file upload object, insert it to DB and give 302', function () {
+      uploadStub.callsArg(2).returns({})
+      claimDocumentUpdateStub.resolves()
+      request
+        .post(VALIDROUTE)
+        .expect(function () {
+          sinon.assert.calledOnce(uploadStub)
+          sinon.assert.calledOnce(fileUploadStub)
+          sinon.assert.calledOnce(claimDocumentUpdateStub)
+        })
+        .expect(302)
+    })
+
+    it('should catch a validation error', function () {
+      uploadStub.callsArg(2).returns({})
+      fileUploadStub.throws(new ValidationError())
+      request
+        .post(VALIDROUTE)
+        .expect(400)
+    })
+
     it('should respond with a 500 if passed invalid document type', function () {
+      uploadStub.callsArg(2).returns({})
       request
-        .get(`${ROUTE}TEST`)
+        .post(`${BASEROUTE}VISIT_CONFIRMATION&claimDocumentId=${CLAIMDOCUMENTID}&eligibilityId=${ELIGIBILITYID}`)
         .expect(500)
```

Comment:
request returning 404 not found, and test is passing anyway. 

This test also states it is passing an invalid document type when VISIT_CONFIRMATION is valid